### PR TITLE
chore(cli): Refactor, add tests, & export for osm-health

### PR DESCRIPTION
**Pull request by johnsonshi**
chore(cli): Refactor, add tests, & export for osm-health

Refactors common code that will be used by osm-health.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

**Description**:

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

### CODE REVIEW

File: cmd/cli/util.go [hunk 1]
```diff
@@ -113,25 +114,21 @@ func getMeshInfoList(restConfig *rest.Config, clientSet kubernetes.Interface) ([
 	return meshInfoList, nil
 }
 
-// getNamespacePods returns a map of controller pods
-func getNamespacePods(clientSet kubernetes.Interface, m string, ns string) map[string][]string {
-	x := make(map[string][]string)
-
+// GetNamespaceOsmControllerPods returns a list of osm-controller pods in the namespace
+func GetNamespaceOsmControllerPods(clientSet kubernetes.Interface, ns string) *corev1.PodList {
```
Comment (shashankram) [inline cmd/cli/util.go:118]:
```suggestion
func GetOSMControllerPods(clientSet kubernetes.Interface, ns string) *corev1.PodList {
```

File: cmd/cli/util.go [hunk 2]
```diff
@@ -113,25 +114,21 @@ func getMeshInfoList(restConfig *rest.Config, clientSet kubernetes.Interface) ([
...
 	listOptions := metav1.ListOptions{
 		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
 	}
-	pods, _ := clientSet.CoreV1().Pods(ns).List(context.TODO(), listOptions)
-
-	for pno := 0; pno < len(pods.Items); pno++ {
-		x["Pods"] = append(x["Pods"], pods.Items[pno].GetName())
+	podList, err := clientSet.CoreV1().Pods(ns).List(context.TODO(), listOptions)
+	if err != nil {
+		return &corev1.PodList{}
```
Comment (shashankram) [inline cmd/cli/util.go:125]: Drop L127 and:
```suggestion
	podList, _:= clientSet.CoreV1().Pods(ns).List(context.TODO(), listOptions)
	return podList
```

The API should return a nil return on error. Alternatively, return an error from this function and let the caller handle it.

File: cmd/cli/util.go (hunk 1 above)
Comment (johnsonshi) [reply cmd/cli/util.go]: @shashankram done :)

**Review by shashankram — APPROVED** [review]

File: cmd/cli/util_test.go [hunk 3]
```diff
@@ -157,6 +163,99 @@ var _ = Describe("Test getting pretty printed output of smi info of a list of me
 	})
 })
 
+func TestGetNamespaceOsmControllerPods(t *testing.T) {
+	testNamespace := "osm-namespace"
+
+	tests := []struct {
+		testName string
+		pods     []*corev1.Pod
+		podNames []string
```
Comment (SanyaKochhar) [inline cmd/cli/util_test.go:172]:
```suggestion
		expectedPodNames []string
```

File: cmd/cli/util_test.go [hunk 4]
```diff
@@ -157,6 +163,99 @@ var _ = Describe("Test getting pretty printed output of smi info of a list of me
...
+						Name:      "some-other-osm-controller-pod",
+						Namespace: "some-other-namespace",
+						Labels: map[string]string{
+							"app": constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-other-application-pod-1",
```
Comment (SanyaKochhar) [inline cmd/cli/util_test.go:206]:
```suggestion
						Name:      "application-pod-1",
```

File: cmd/cli/util_test.go [hunk 5]
```diff
@@ -157,6 +163,99 @@ var _ = Describe("Test getting pretty printed output of smi info of a list of me
...
+						Namespace: testNamespace,
+						Labels: map[string]string{
+							"app": "myapp",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-other-application-pod-2",
```
Comment (SanyaKochhar) [inline cmd/cli/util_test.go:215]:
```suggestion
						Name:      "application-pod-2",
```

File: cmd/cli/util_test.go [hunk 6]
```diff
@@ -3,14 +3,20 @@ package main
 import (
 	"context"
 	"fmt"
+	"sort"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/runtime"
+
+	corev1 "k8s.io/api/core/v1"
+
```
Comment (SanyaKochhar) [inline cmd/cli/util_test.go:12]: nit: I think the newlines can be removed as these are all external imports so can be grouped together

File: cmd/cli/util_test.go [hunk 7]
```diff
@@ -157,6 +163,99 @@ var _ = Describe("Test getting pretty printed output of smi info of a list of me
...
+			podList := GetOSMControllerPods(fakeClientSet, testNamespace)
+
+			assert.Equal(len(test.podNames), len(podList.Items))
+			actualPodNames := make([]string, len(podList.Items))
+			for i, pod := range podList.Items {
+				actualPodNames[i] = pod.Name
+			}
+			sort.Strings(test.podNames)
+			sort.Strings(actualPodNames)
+			assert.Equal(test.podNames, actualPodNames)
```
Comment (SanyaKochhar) [inline cmd/cli/util_test.go:254]: I think tassert may have a method for this, try assert.ElementsMatch? https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

**Review by jaellio — APPROVED** [review]

**Pull request merged**